### PR TITLE
添加页面排序

### PR DESCRIPTION
**Pull request by aqssxlzc**
添加Page列表按照数字排序功能
PageList sort by number in filename.

### CODE REVIEW

**Comment by tak2hu** [comment]
every `data/icon` is deleted.

other than that, thanks for adding this! 
(haven’t tested it yet myself, later)

File: ballontranslator/ui/mainwindow.py [hunk 1]
```diff
@@ -38,7 +39,23 @@ def __init__(self, *args, **kwargs) -> None:
...
+            if re.match("/d+",f):
+                return int(f)
+        return item_str
+    def __lt__(self, other) -> bool:
+        print(self.getNum(self.text()),self.getNum(other.text()))
+        try:
+            self.getNum(self.text())<self.getNum(other.text())
+        except Exception:
+            return super().__lt__(other)
+        return super().__lt__(other)
```
Comment (dmMaze) [inline ballontranslator/ui/mainwindow.py]: 这里try里面没有返回值
而且下面例子也没排对
image

**Comment by tak2hu** [comment]
also adding something like
```
        tmp_pages = list(self.pages)
        tmp_pages.sort()
        
        for page in tmp_pages:
            pagename = page
            blklist = self.pages[page]
            imgpath = os.path.join(self.directory, pagename)
```

at [imgtrans_proj.py](https://github.com/dmMaze/BallonsTranslator/blob/657f18af602b738c88495f7aaf6078f0139eaae6/ballontranslator/ui/imgtrans_proj.py#L308) in `dump_doc()` would be nice when sending a docx with ordered pages to a proofreader/translator 

(PS: I use this pr and above, it really helps with 50+ pages, but I can't use `import from doc` because it's bugged on linux with python 3.11, so I type the changes from a proofreader manually)

**Comment by dmMaze** [comment]
> also adding something like
> 
> ```
>         tmp_pages = list(self.pages)
>         tmp_pages.sort()
>         
>         for page in tmp_pages:
>             pagename = page
>             blklist = self.pages[page]
>             imgpath = os.path.join(self.directory, pagename)
> ```
> 
> at [imgtrans_proj.py](https://github.com/dmMaze/BallonsTranslator/blob/657f18af602b738c88495f7aaf6078f0139eaae6/ballontranslator/ui/imgtrans_proj.py#L308) in `dump_doc()` would be nice when sending a docx with ordered pages to a proofreader/translator
> 
> (PS: I use this pr and above, it really helps with 50+ pages, but I can't use `import from doc` because it's bugged on linux with python 3.11, so I type the changes from a proofreader manually)

then it would be more convenient  to sort pages directly in the loading function 1e03a003c0158974ba3fe76e0908c3e62b759950, thus I'll close this pr since it only dealt with the qt view 
I'll look into the doc import from problem with linux

**Pull request closed without merging**